### PR TITLE
BUG-108475 Management pack v3 auth

### PR DESCRIPTION
**Pull request by topolyai5**

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/repository/ManagementPackRepository.java [hunk 1]
```diff
@@ -3,18 +3,22 @@
...
 import com.sequenceiq.cloudbreak.aspect.HasPermission;
+import com.sequenceiq.cloudbreak.aspect.organization.OrganizationResourceType;
+import com.sequenceiq.cloudbreak.domain.ManagementPack;
 import com.sequenceiq.cloudbreak.service.EntityType;
+import com.sequenceiq.cloudbreak.validation.OrganizationPermissions.Resource;
 
 @EntityType(entityClass = ManagementPack.class)
-@Transactional(Transactional.TxType.REQUIRED)
+@Transactional(TxType.REQUIRED)
 @HasPermission
```
Comment (attilapalfi92) [inline core/src/main/java/com/sequenceiq/cloudbreak/repository/ManagementPackRepository.java]: remove HasPermission and add DisableHasPermission

File: core/src/main/resources/schema/app/20180814112145_BUG-108475_mpack_auth.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+-- // BUG-108475 mpack auth
+-- Migration SQL that makes the change goes here.
+
+ALTER TABLE managementpack ALTER COLUMN owner DROP NOT NULL;
+ALTER TABLE managementpack ALTER COLUMN account DROP NOT NULL;
+ALTER TABLE managementpack DROP CONSTRAINT IF EXISTS uk_managementpack_account_name;
+
+UPDATE managementpack SET organization_id = -1 where organization_id IS NULL;
```
Comment (mhmxs) [inline core/src/main/resources/schema/app/20180814112145_BUG-108475_mpack_auth.sql]: Will fail

File: core/src/main/java/com/sequenceiq/cloudbreak/repository/OrganizationResourceRepository.java [hunk 3]
```diff
@@ -6,19 +6,25 @@
...
+import com.sequenceiq.cloudbreak.aspect.DisabledBaseRepository;
 import com.sequenceiq.cloudbreak.aspect.organization.CheckPermissionsByOrganization;
+import com.sequenceiq.cloudbreak.aspect.organization.CheckPermissionsByOrganizationId;
 import com.sequenceiq.cloudbreak.domain.organization.Organization;
 import com.sequenceiq.cloudbreak.domain.organization.OrganizationAwareResource;
 
 @NoRepositoryBean
 @DisableHasPermission
-public interface OrganizationResourceRepository<T extends OrganizationAwareResource, ID extends Serializable> extends BaseRepository<T, ID> {
+public interface OrganizationResourceRepository<T extends OrganizationAwareResource, ID extends Serializable> extends DisabledBaseRepository<T, ID> {
```
Comment (attilapalfi92) [inline core/src/main/java/com/sequenceiq/cloudbreak/repository/OrganizationResourceRepository.java]: If you want to extend DisabledBaseRepository, DisabledBaseRepository's methods must be annotated like BaseRepository's.
like:
@CheckPermissionsByTarget(action = WRITE, targetIndex = 0) and etc

Comment (topolyai5) [reply core/src/main/java/com/sequenceiq/cloudbreak/repository/OrganizationResourceRepository.java]: I've changed back to BaseRepository.

**Pull request merged**